packages/common/src/hardhat/tasks/identifiers.ts

```diff
@@ -27,6 +28,47 @@ function isString(input: string | null): input is string {
   return typeof input === "string";
 }
 
+async function checkIfIdentifiersAreSupported(
+  identifiersToCheck: string[],
+  web3: Web3,
+  whitelistToCheck: Contract,
+  hre: CombinedHRE
+): Promise<boolean[]> {
+  const networkId = await web3.eth.getChainId();
+  console.log(
+    `Checking whitelist status of ${identifiersToCheck.length} identifiers on the whitelist @ ${whitelistToCheck.options.address} on network ID ${networkId} `
+  );
+  const whitelistedIdentifiers: boolean[] = [];
+  if (MULTICALL2_DEPLOYMENTS[networkId]) {
+    const multicallAddress = MULTICALL2_DEPLOYMENTS[networkId];
+    console.log(`Using multicall contract @ ${multicallAddress} to reduce web3 requests`);
+    const Multicall2 = hre.getContract("Multicall2");
+    const multicall = new web3.eth.Contract(Multicall2.abi, multicallAddress);
+    const calls = [];
+    for (const id of identifiersToCheck) {
+      calls.push({
+        target: whitelistToCheck.options.address,
+        callData: whitelistToCheck.methods.isIdentifierSupported(id).encodeABI(),
+      });
+    }
+    const result = await multicall.methods.aggregate(calls).call();
+    for (const _result of result.returnData) {
+      // Multicall contract returns results as bytes, so we need to check return value against bytes representation of
+      // Boolean value.
+      whitelistedIdentifiers.push(_result === "0x0000000000000000000000000000000000000000000000000000000000000001");
+    }
+  } else {
+    console.log(
+      `No multicall contract found for network ${networkId}, submitting ${identifiersToCheck.length} web3 requests, sit tight`
+    );
+    for (const id of identifiersToCheck) {
```

Comment:
Are these intentionally done in series to avoid overloading the node? I only ask because it could be faster if we used an `await Promise.all(identifierToCheck.map(`.

Reply:
Ah yes we should do that!